x/stakeibc/keeper/send_host_zone_unbondings_test.go

```diff
@@ -0,0 +1,164 @@
+			Weight:        wgt_val2,
+		},
+	}
+
+	delegationAccount := stakeibc.ICAAccount{
+		Address: delegationAddr,
+		Target:  stakeibc.ICAAccountType_DELEGATION,
+	}
+
+	hostZone := stakeibc.HostZone{
+		ChainId:           "GAIA",
+		HostDenom:         "uatom",
+		Bech32Prefix:      "cosmos",
+		Validators:        validators,
+		DelegationAccount: &delegationAccount,
+	}
+
+	// list of epoch unbonding records
+	epochUnbondingRecords := []recordtypes.EpochUnbondingRecord{
+		recordtypes.EpochUnbondingRecord{
+			Id:                   1,
+			UnbondingEpochNumber: 1,
+			HostZoneUnbondings:   make(map[string]*recordtypes.HostZoneUnbonding),
+		},
+	}
+
+	// for each epoch unbonding record, add a host zone unbonding record and append the record
+	for _, epochUnbondingRecord := range epochUnbondingRecords {
+		// note: we're using the same hostZoneUnbonding for each epoch unbonding record in this test
+		epochUnbondingRecords[0].HostZoneUnbondings["GAIA"] = &recordtypes.HostZoneUnbonding{
+			Amount:        amt_to_unbond,
+			Denom:         "uatom",
+			HostZoneId:    "GAIA",
+			UnbondingTime: unbonding_time, // 2022-08-12T19:51
+			Status:        recordtypes.HostZoneUnbonding_BONDED,
+		}
+
+		s.App.RecordsKeeper.AppendEpochUnbondingRecord(s.Ctx, epochUnbondingRecord)
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	return SendHostZoneUnbondingTestCase{
+		amt_to_unbond:           amt_to_unbond,
+		host_zone:               hostZone,
+		epoch_unbonding_records: epochUnbondingRecords,
+		lightClientTime:         light_client_time, // 2022-08-13T19:51, 1d after the unbonding time
+	}
+}
+
+func (s *KeeperTestSuite) TestSendHostZoneUnbondingSuccessful() {
+	tc := s.SetupSendHostZoneUnbonding()
+
+	msgs, totalAmtToUnbond, err := s.App.StakeibcKeeper.SendHostZoneUnbondings(s.Ctx, tc.host_zone)
+	s.Require().Nil(err)
+
+	// function to get total wgt of all validators on a host zone
+	getTotalWgt := func(hostZone stakeibc.HostZone) sdk.Dec {
+		totalWgt := sdk.ZeroDec()
+		for _, validator := range hostZone.Validators {
+			totalWgt = totalWgt.Add(sdk.NewDec(int64(validator.Weight)))
+		}
+		return totalWgt
+	}
+	totalWgt := getTotalWgt(tc.host_zone)
+	amtToUnbond := sdk.NewDec(int64(tc.amt_to_unbond))
+	// due to erratic rounding and random validator ordering, we can only guarantee that at least one of the undelegations is correct
+	a := strings.Contains(msgs[1].String(), (sdk.NewDec(int64(tc.host_zone.Validators[1].Weight)).Quo(totalWgt).Mul(amtToUnbond)).TruncateInt().String())
+	b := strings.Contains(msgs[0].String(), (sdk.NewDec(int64(tc.host_zone.Validators[1].Weight)).Quo(totalWgt).Mul(amtToUnbond)).TruncateInt().String())
+	c := strings.Contains(msgs[1].String(), (sdk.NewDec(int64(tc.host_zone.Validators[0].Weight)).Quo(totalWgt).Mul(amtToUnbond)).TruncateInt().String())
+	d := strings.Contains(msgs[0].String(), (sdk.NewDec(int64(tc.host_zone.Validators[0].Weight)).Quo(totalWgt).Mul(amtToUnbond)).TruncateInt().String())
```

Comment:
Might be a little easier to follow this is if was broken out into variables
```suggestion
         actualUnbondMsg1 := actualUnbondMsgs[0].String()
         actualUnbondMsg2 := actualUnbondMsgs[1].String()
         val1Fraction := sdk.NewDec(int64(tc.host_zone.Validators[0].Weight)).Quo(totalWgt)
         val2Fraction := sdk.NewDec(int64(tc.host_zone.Validators[1].Weight)).Quo(totalWgt)
         val1UnbondAmt := valFraction1.Mul(actualAmtToUnbond).TruncateInt().String()
         val2UnbondAmt := valFraction2.Mul(actualAmtToUnbond).TruncateInt().String()
         
         val1Unbonded := strings.Contains(actualUnbondMsg1, val1UnbondAmt)
         val2Unbonded := strings.Contains(actualUnbondMsg2, val2UnbondAmt)
        
        s.Require().True(val1Unbonded || val2Unbonded)
```